extras/task_yaml_path_checker.py

```diff
@@ -0,0 +1,93 @@
+                    f"Collection '{collection_id}' referenced by study '{study}' was not found in yaml files")
+            else:  # check the collection
+                task_ids = collections[collection_id].task_ids
+                tasks: Dict[str, RawTaskParams] = meta.read_tasks()
+                for task_id in task_ids:
+                    if task_id not in tasks:
+                        raise PathError(f"Task '{task_id}' referenced by collection '{collection_id}' "
+                                        f"was not found in yaml files")
+                    else:  # check the task
+                        task = tasks[task_id]
+                        instructions: Dict[str, InstructionArgs] = meta.read_instructions()
+                        instruction_id = task.instruction_id
+                        if instruction_id and not instruction_id in instructions:
+                            raise PathError(f"Instruction '{instruction_id}' referenced by task '{task_id}' "
+                                            f"was not found in the yaml files. "
+                                            f"Task '{task_id}' is referenced by collection '{collection_id}'.")
+                        stimuli: Dict[str, StimulusArgs] = meta.read_stimuli()
+                        stim_id = task.stimulus_id
+                        if stim_id not in stimuli:
+                            raise PathError(f"Stimulus '{stim_id}' referenced by task '{task_id}' "
+                                            f"was not found in the yaml files. "
+                                            f"Task '{task_id}' is referenced by collection '{collection_id}'.")
+                        device_ids = task.device_id_array
+                        devices: Dict[str, DeviceArgs] = meta.read_devices()
+                        for dev_id in device_ids:
+                            if dev_id not in devices:
+                                raise PathError(
+                                    f"Device '{dev_id}' referenced by task '{task_id}' "
+                                    f"was not found in the yaml files. Task '{task_id}' is referenced by collection"
+                                    f" '{collection_id}'.")
+                            device: DeviceArgs = devices[dev_id]
+                            sensor_ids = device.sensor_ids
+                            sensors: Dict[str, SensorArgs] = meta.read_sensors()
+                            for sensor_id in sensor_ids:
+                                if sensor_id not in sensors:
+                                    raise PathError(f"Sensor '{sensor_id}' referenced by device '{dev_id}' "
+                                                    f"was not found in the yaml files. "
+                                                    f"Device '{dev_id}' is referenced by task '{task_id}', "
+                                                    f"and '{task_id}' is referenced by collection '{collection_id}'.")
+
```

Comment:
I'm fine with this at the moment, but part of me feels like it could be written more concisely recursively. Maybe this is more easily understandable though.

Another option could be if we add a `validate_references()` to each XXXArgs object that handles the check in more more modular way?